### DISCUSSION TITLE
Integration with Deno & Bundler

### DISCUSSION THREAD

**Discussion by tarsupin** [excaliburjs/Excalibur#2060] [Q&A] [answered]
With Deno now in a full and stable release, and since it uses Typescript natively, it seems like an appropriate fit for Excalibur. While Node does have a more established community, those of us who have moved to Deno just can't ever go back.

Nearly everything is compatible between Node and Deno, so I suspect this would be a near-trivial addition, and one that is very sought after for those looking into Typescript-based engines.

**Comment by kamranayub** [comment]
In fact we were just discussing evaluating Deno to replace our Node bundling/build pipeline 👍 

I am not well-versed enough yet to know what libraries need to do to support Deno. Do you have an idea for what we'd need to change with the output we generate? Keep in mind Excalibur is client-side only and AFAIK Deno is a server, so I'm not quite sure where Excalibur fits in there.

**edit:** Moved to Discussions because I am not sure yet if there's anything needed here.

**Comment by tarsupin** [comment]
Node is also strictly server-side, but it's also your entire installation process and presumably used for the bundling process for Typescript to Javascript. I'm suggesting Deno as a drop-in replacement, or at least an alternative, to do the same. This way we can choose whether or not we want to install Node & npm.

So to answer your question, I assume next to nothing will be needed. Just provide details in the Getting Started section for setup and bundling with Deno so that anyone can use it with a copy/paste.

**Reply by kamranayub** [reply]
I see, so this would be for some kind of Deno frontend framework or project that uses Deno's bundling/module system instead of node/npm. Like providing commands for a npm vs. yarn install etc. I think I have to see an example of what this brave new world looks like 😄

**Reply by kamranayub** [reply]
**edit:** we updated docs

**Comment by kamranayub** [comment, marked as answer]
Hey! Thanks to @return180bpm we now have guidance and documentation for using Deno bundle:

https://excaliburjs.com/docs/bundlers#deno

Thanks for getting the discussion started!
